emily/handler/src/api/handlers/deposit.rs

```diff
@@ -439,19 +440,32 @@ pub async fn update_deposits(
         // in order to enforce added stability to the API during a reorg.
         let api_state = accessors::get_api_state(&context).await?;
         api_state.error_if_reorganizing()?;
+
+        let is_trusted_key = context.settings.trusted_reorg_api_key == api_key;
+        // Signers are only allowed to update deposits to the accepted state.
+        if !is_trusted_key {
+            let is_unauthorized = body
+                .deposits
+                .iter()
+                .any(|deposit| deposit.status != Status::Accepted);
+
+            if is_unauthorized {
+                return Err(Error::Forbidden);
+            }
```

Comment:
Should further limit the signers update to state transition from pending to accepted? Otherwise it could be possible that a signer change a confirmed deposit back to accepted. I looked around but I couldn't find anything preventing this, but maybe I'm missing something.